src/etos_test_runner/lib/executor.py

```diff
@@ -144,13 +144,17 @@ def _checkout_tests(self, test_checkout, workspace):
         self.logger.info("Checkout script:\n %s", checkout.read_text())
 
         signal.signal(signal.SIGALRM, _test_checkout_signal_handler)
-        signal.alarm(60)
-        try:
-            success, output = self._call(
-                ["/bin/bash", str(checkout)], shell=True, wait_output=False
-            )
-        finally:
-            signal.alarm(0)
+        success = False
```

Comment:
We should probably re-create the workspace directory as well before retrying since git cannot checkout in a non-empty repository. This can happen if one checkout succeeded and another failed, and we then retry.